spec/requests/courses_spec.rb

```diff
@@ -1,19 +1,73 @@
 require 'helpers/rails_helper'
+require 'helpers/api_helper'
 
 RSpec.describe "Courses", type: :request do
   include_context "with authenticated requests"
-  let(:user) { create(:user) }
+  let(:student) { create(:user) }
+  let(:teacher) { create(:user, :teacher) }
   let!(:course) { create(:course) }
 
-  before do
```

Comment:
In this set of specs, we can
`before do
  authenticate(user)
end
`
and in each test case, `let(:user) { create(:user) }`, or `let(:user) { create(:user, :teacher) }`. The spec will used the "user" specified in the scope most specific to the spec.

We may also consider making different factories entirely for student and teacher, instead of having teacher be a trait.